### DISCUSSION TITLE
Migrating From PhysX SDK 4.0 to 5.1

### DISCUSSION THREAD

**Discussion by JiyunRuan** [NVIDIA-Omniverse/PhysX#114] [Q&A]
Hi
I used the custom constraint to add a new joint with PhysX4.1. After I updated the engine to 5.1, the joint behavior goes wrong. I didn't change the code of logic. I checked the web about migrating from SDK 4.0 to 5.1 and there is no info about PxRigidbody/PxRigidDynamic or custom constraint. Do you have any information about these? or is there no difference between 4.1 and 5.1? 
Best

**Comment by preist-nvidia** [comment]
Hi @JiyunRuan,

We had other users of custom constraints upgrading from 4.1. to 5.1 without issues and nothing should have changed in terms of behavior.

Are you sure that no other simulation parameters changed when you ported that could affect constraint behavior?

Philipp

**Comment by Justin19831203** [comment]
There are more limitations in PhysX5 than PhysX. For example, "PxArticulationJointReducedCoordinate::setMotion() not allowed while the articulation is in a scene. Call will be ignored." "PxArticulationJointReducedCoordinate::setLimitParams() not allowed while simulation is running. Call will be ignored." These make it impossible for our project to be upgrade to PhysX5.
